File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry= 1;
        for(int i=digits.size()-1; i>=0; i--){
            int sum= digits[i]+carry;
            carry= sum/10;
            if(sum<10)
                digits[i]=sum;
            else
                digits[i]=0;
        }
        if(carry)
            digits.insert(digits.begin(),1);
        return digits;
    }
};

File: 0066-plus-one/0066-plus-one.py
# TC- O(n)
# SC- O(1)

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        n = len(digits)
        for i in range(n-1, -1, -1):
            temp_sum = digits[i]+carry
            digits[i] = temp_sum%10
            carry = temp_sum//10
        if carry == 1:
            digits.insert(0, 1)
        return digits